src/sentry/models/activity.py

```diff
@@ -32,18 +32,29 @@
 
 class ActivityManager(BaseManager["Activity"]):
     def get_activities_for_group(self, group: Group, num: int) -> Sequence[Group]:
+        from sentry.issues.priority import PRIORITY_LEVEL_TO_STR
+
         activities = []
         activity_qs = self.filter(group=group).order_by("-datetime")
+        initial_priority = None
 
         if not features.has("projects:issue-priority", group.project):
-            activity_qs.exclude(type=ActivityType.SET_PRIORITY.value)
+            activity_qs = activity_qs.exclude(type=ActivityType.SET_PRIORITY.value)
+        else:
+            initial_priority = (
+                PRIORITY_LEVEL_TO_STR[
+                    group.get_event_metadata()["initial_priority"]
+                ]  # inital_priority is only available when the feature flag is on
```

Comment:
It also won't be available until the migration is run. We should do a safe get here: `group.get_event_metadata().get("initial_priority")`

Reply:
Good catch!